torch/_inductor/ir.py

```diff
@@ -2011,15 +2064,15 @@ def reindex(index, sort_index):
         return results
 
 
-def is_storage_and_layout(x):
+def is_storage_and_layout(x: object) -> bool:
```

Comment:
IRNode ?

Reply:
Should this be a TypeGuard or TypeIs typing_extension?

Reply:
The problem is that it doesn't really resolve to a type (or set of types) as much as a concept.